From 4ad702485dfdfd7a259c90217bdd2f393674251d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Joachim=20Mairb=C3=B6ck?= <j.mairboeck@gmail.com>
Date: Sun, 5 Jan 2025 23:42:32 +0100
Subject: Haiku build support

Set installation paths to Haiku conventions.

Set LIBRARY_PATH for tests.

diff --git a/Build.PL b/Build.PL
index 3ec4ef8..488b985 100644
--- a/Build.PL
+++ b/Build.PL
@@ -87,6 +87,15 @@ if ( $^O =~ /mswin32/i ) {
         unlink catfile( $libdir, $target );
     }
 }
+elsif ( $^O =~ /haiku/i ) {
+    $libdir =~ s/\bbin\b/lib/;
+    $hdrdir = $libdir;
+    $hdrdir =~ s!\blib\b!develop/headers!;
+    if ( $Config{archname} =~ /BePC/ ) {
+        $libdir .= "/x86";
+        $hdrdir .= "/x86";
+    }
+}
 else {
     if ( $Config{archname} =~ /^x86_64|^ppc64|^s390x|^aarch64|^riscv64/ ) {
         $libdir =~ s/\bbin\b/lib64/;
diff --git a/inc/MyBuilder.pm b/inc/MyBuilder.pm
index 8c6f6f9..926ddf4 100644
--- a/inc/MyBuilder.pm
+++ b/inc/MyBuilder.pm
@@ -385,6 +385,10 @@ sub ACTION_test {
         my $oldlibpath = $ENV{LIBPATH} || '/lib:/usr/lib';
         $ENV{LIBPATH} = catdir($self->blib, "usrlib").":$oldlibpath";
     }
+    elsif ($^O =~ /haiku/i) {
+        my $oldlibpath = $ENV{LIBRARY_PATH} || '/boot/system/lib:/boot/system/lib/x86';
+        $ENV{LIBRARY_PATH} = catdir($self->blib, "usrlib").":$oldlibpath";
+    }
     elsif ($^O =~ /cygwin/i) {
         # cygwin uses windows lib searching (PATH instead of LD_LIBRARY_PATH)
         my $oldpath = $ENV{PATH};
-- 
2.45.2

